DEV.md

```diff
@@ -0,0 +1,72 @@
+# Developer Notes
```

Comment:
per https://github.com/opentracing/opentracing.github.io/issues/10, point to `opentracing.io` for general background on the project, semantics, etc.